src/handler/sessions/session.rs

```diff
@@ -265,3 +328,17 @@ impl Session {
         Ok((packet, session))
     }
 }
+
+enum MostRecentEnr<'a> {
+    Challenge(&'a Enr),
```

Comment:
the `challenge` variable in the code a few lines above is only used to get the `data`. So,  most likely you can destructure the `Challenge`, use the `data` where needed, and add the `Enr` as value here to avoid the `unwrap`/`expect`

Reply:
then I would have to reconstruct the Challenge for this error type which I wanted to avoid changing, which feels counter intuitive, but I'll change the error

https://github.com/sigp/discv5/blob/715fc822dfd7f9d2fb1eb43cf4b8e771768c283b/src/error.rs#L21

Reply:
oh no I remember why this is the best way the way it is already, feel free to change it though. the challenge needs to be reinserted.

https://github.com/sigp/discv5/blob/715fc822dfd7f9d2fb1eb43cf4b8e771768c283b/src/handler/mod.rs#L820

Reply:
this is what it would look like https://github.com/divagant-martian/discv5/commit/d98b8d1aec28ca833bdfa2961d043ade23aebdbd